newdevtech/newdev_api
# #166457282 Implement user login

#### What does this PR do?
- This PR implements the user login feature for our app

#### Description of Task to be completed?
- add middleware for user login
- add `route` for user login
- add `controller` for user login
- delete redundant index files

#### How should this be manually tested?
- send a `POST` request to `/user/auth/login` with the required `valid` params:
    - `usernameOrEmail` and `password`
- you should receive a successful response message with `JWT token`.

#### What are the relevant pivotal tracker stories?
[#166457282](https://www.pivotaltracker.com/story/show/166457282)
